components/tidb_query_vec_executors/src/runner.rs

```diff
@@ -469,13 +430,46 @@ impl<SS: 'static> BatchExecutorsRunner<SS> {
             }
 
             // Grow batch size
-            if batch_size < BATCH_MAX_SIZE {
-                batch_size *= BATCH_GROW_FACTOR;
-                if batch_size > BATCH_MAX_SIZE {
-                    batch_size = BATCH_MAX_SIZE
-                }
-            }
+            grow_batch_size(&mut batch_size);
+        }
+    }
+
+    pub fn handle_streaming_request(
+        &mut self,
+    ) -> Result<(Option<(StreamResponse, IntervalRange)>, bool)> {
+        let mut warnings = self.config.new_eval_warnings();
+
+        let (mut record_len, mut is_drained) = (0, false);
+        let mut chunk = Chunk::default();
+        let mut ctx = EvalContext::new(self.config.clone());
+        // record count less than batch size and is not drained
+        while record_len < self.streaming_batch_limit && !is_drained {
+            let mut current_chunk = Chunk::default();
+            let (drained, len) = self.internal_handle_request(
+                true,
+                self.stream_batch_size
+                    .min(self.streaming_batch_limit - record_len),
+                &mut current_chunk,
+                &mut warnings,
+                &mut ctx,
+            )?;
+            chunk
+                .mut_rows_data()
+                .extend_from_slice(current_chunk.get_rows_data());
```

Comment:
If the `len` exceeds the `stream_row_limit`, then the chunks should be split into two responses or we could just inform the executor to fetch the `stream_row_limit` rows each time.